symfony/plugins/orangehrmAdminPlugin/Service/SystemUserService.php

```
@@ -0,0 +1,322 @@
+ */
+
+namespace OrangeHRM\Admin\Service;
+
+use OrangeHRM\Admin\Dao\SystemUserDao;
+use OrangeHRM\Core\Exception\ServiceException;
+use OrangeHRM\Core\Utility\PasswordHash;
+use OrangeHRM\Entity\SystemUser;
+use OrangeHRM\SecurityAuthentication\Dto\UserCredential;
+
+class SystemUserService {
+    /**
+     * @var SystemUserDao|null
+     */
+    protected ?SystemUserDao $systemUserDao = null;
+    
+    /** @property PasswordHash $passwordHasher */
+    private ?PasswordHash $passwordHasher = null;
+
+    /**
+     * @return SystemUserDao
+     */
+    public function getSystemUserDao(): ?SystemUserDao
+    {
+        if (empty($this->systemUserDao)) {
+            $this->systemUserDao = new SystemUserDao();
+        }
+        return $this->systemUserDao;
+    }
+
+    /**
+     * @param SystemUserDao $systemUserDao
+     */
+    public function setSystemUserDao(SystemUserDao $systemUserDao): void
+    {
+        $this->systemUserDao = $systemUserDao;
+    }
+
+    /**
+     * @return PasswordHash
+     */
+    public function getPasswordHasher(): PasswordHash
+    {
+        if (empty($this->passwordHasher)) {
+            $this->passwordHasher = new PasswordHash();
+        }        
+        return $this->passwordHasher;
+    }
+
+    public function setPasswordHasher(PasswordHash $passwordHasher): void
+    {
+        $this->passwordHasher = $passwordHasher;
+    }
+
+    
+    /**
+     * Save System User
+     * 
+     * @param SystemUser $systemUser
+     * @return void
+     */
+    public function saveSystemUser(SystemUser $systemUser,$changePassword = false){
+        
+        try {
+            if ($changePassword) {
+                $systemUser->setUserPassword($this->hashPassword($systemUser->getUserPassword()));
+            }
+
+            return $this->getSystemUserDao()->saveSystemUser($systemUser);
+            
+        } catch (\Exception $e) {
+            throw new ServiceException($e->getMessage(),$e->getCode(),$e);
```

This also not a good way to handle the exception.. we should catch exception in top layers ( Action and APIs)
here should throw an exception if something wrong